packages/block-editor/src/components/block-list/zoom-out-separator.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__unstableMotion as motion,
+	__unstableAnimatePresence as AnimatePresence,
+} from '@wordpress/components';
+import { useReducedMotion } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+export function ZoomOutSeparator( {
+	clientId,
+	rootClientId,
+	position = 'top',
+} ) {
+	const {
+		isZoomOut,
+		sectionRootClientId,
+		sectionClientIds,
+		blockInsertionPoint,
+		blockInsertionPointVisible,
+	} = useSelect( ( select ) => {
+		const {
+			getSettings,
+			__unstableGetEditorMode,
+			getBlockInsertionPoint,
+			getBlockOrder,
+			isBlockInsertionPointVisible,
+		} = unlock( select( blockEditorStore ) );
+
+		if ( __unstableGetEditorMode() === 'zoom-out' ) {
```

Comment:
If we pass `editorMode` from the parent component, we can avoid creating store subscriptions for non-zoomed-out modes.

## How?

If you return early in the `mapSelect` function before calling `select( blockEditorStore )`, no store subscription will be created.

Reply:
Something like [this](https://github.com/WordPress/gutenberg/pull/63896/commits/675d7da09d3ba010d7d98842bda64a919172e2a0)?

Reply:
Thanks, @MaggieCabrera!